tests/filtertransformers/test_mongo.py

```diff
@@ -260,46 +262,73 @@ def test_operators(self):
         with pytest.raises(VisitError):
             self.transform('"some string" > "some other string"')
 
-    def test_filtering_on_relationships(self):
+    def test_filtering_on_relationships(self, mapper):
         """Test the nested properties with special names
         like "structures", "references" etc. are applied
         to the relationships field.
 
         """
+        from optimade.filtertransformers.mongo import MongoTransformer
+
+        t = MongoTransformer(mapper=mapper("StructureMapper"))
+        p = LarkParser(version=self.version, variant=self.variant)
 
-        assert self.transform('references.id HAS "dummy/2019"') == {
+        assert t.transform(p.parse('references.id HAS "dummy/2019"')) == {
             "relationships.references.data.id": {"$in": ["dummy/2019"]}
         }
 
-        assert self.transform('structures.id HAS ANY "dummy/2019", "dijkstra1968"') == {
+        assert t.transform(
+            p.parse('structures.id HAS ANY "dummy/2019", "dijkstra1968"')
+        ) == {
             "relationships.structures.data.id": {"$in": ["dummy/2019", "dijkstra1968"]}
         }
 
-        assert self.transform('structures.id HAS ALL "dummy/2019", "dijkstra1968"') == {
+        assert t.transform(
+            p.parse('structures.id HAS ALL "dummy/2019", "dijkstra1968"')
+        ) == {
             "relationships.structures.data.id": {"$all": ["dummy/2019", "dijkstra1968"]}
         }
 
-        assert self.transform('structures.id HAS ONLY "dummy/2019"') == {
+        assert t.transform(p.parse('structures.id HAS ONLY "dummy/2019"')) == {
             "$and": [
                 {"relationships.structures.data": {"$size": 1}},
                 {"relationships.structures.data.id": {"$all": ["dummy/2019"]}},
             ]
         }
 
-        assert self.transform(
-            'structures.id HAS ONLY "dummy/2019" AND structures.id HAS "dummy/2019"'
+        assert t.transform(
+            p.parse(
+                'structures.id HAS ONLY "dummy/2019" AND structures.id HAS "dummy/2019"'
+            )
         ) == {
             "$and": [
                 {
                     "$and": [
-                        {"relationships.structures.data": {"$size": 1}},
+                        {
+                            "relationships.structures.data": {
+                                "$size": 1,
+                            }
+                        },
                         {"relationships.structures.data.id": {"$all": ["dummy/2019"]}},
                     ]
                 },
                 {"relationships.structures.data.id": {"$in": ["dummy/2019"]}},
             ],
         }
 
+    def test_other_provider_fields(self, mapper):
+        """Test that fields from other providers generate
+        queries that treat the value of the field as `null`.
```

Comment:
I do not see how this test does what is described in the description. Isn't the transformation of the query the same as for a field that is supported by this provider?

Reply:
This test is in contrast to a provider field for the current provider, i.e. `_exmpl_provider_field` which returns an error. Here, the `_other_` prefix is checked against the list of known providers, and must be passed through as a query on the missing field. e.g. `elements HAS "H" and _other_provider_field > 1` should return no results. Our handling of null/missing fields should deal with this for more complicated queries